.ci/README.md

```diff
@@ -1,63 +1,87 @@
+**If you want to run the terraform tests on the generated code**, you should run this command:
 
-## `merge-prs`
+```
+fly execute -t main --config .ci/unit-tests/test-terraform.yml --input terraform=/tmp/terraform-output  --include-ignored
+```
 
-The fourth stage's goal is to merge `magic-modules` PRs after they have been approved and all the downstream PRs have been merged.
+**If you want to run the terraform tests on the existing submodule**, you should run this command:
 
-### `merge-and-update`
+```
+fly execute -t main --config .ci/unit-tests/task.yml --input magic-modules=. --include-ignored
+```
 
-`merge-and-update` takes two inputs:
-* The approved PR, after all downstream PRs have been merged.
-* This CI repository.
+# CI For MagicModules Developers
+If you develop MagicModules (code generation features), here are the things you'll want to know.
 
-It then runs `magic-modules/merge.yml`, which declares one output:
+## Adding Tests
+You can easily add things to the test suites.  The version of the `.yml` and `.sh` files in this subdirectory which are run when you call `fly execute` are the versions at HEAD in your local copy of the repository.  If you write new tests which require some setup, you can just add them to the shell scripts which are already being executed - if the overall shell script exits with a nonzero code, the task will be marked as failing.
 
-* The `magic-modules` PR repo after it has been updated to be ready to merge.
+# CI For CI Developers
+If you're working on enhancing these CI tools, you'll need to know about the structure of the pipeline and about how to develop and test.
 
-#### A note on tracking submodules
-This job sets the submodules back to tracking their downstream repositories on `master`.  It updates the submodules to point to the most recent commit on `master`.  This may not be ideal - other commits may have been made to `master` since the downstream PR was merged - however, this is the best way to ensure that we do not go backwards.  Imagine the following situation:
+## Resources, Jobs, and Tasks
+Concourse has three primitives - Resources, Jobs, and Tasks.
+
+* Resources are things which represent state external to Concourse.  A git repository would be a resource, or a GitHub repo's pull request queue, or a Docker Hub page.  Resources have versions, which, taken together with the resource definition, represent a single immutable piece of state.  If the state changes, the version also changes.  For this reason, versions are usually hashes of the state: A commit hash would be a version, or a PR number / merge SHA pair, or a container SHA.
```

Comment:
optional nit- suggested phrasing: "Resources represent state external to Concourse, such as a git repository, a GitHub repo's pull request queue, or a Docker Hub page."

Reply:
That's cleaner - I don't know what I was doing.  :)